### PR TITLE
DATA-560 logstash gcp folder

### PR DESCRIPTION
**Pull request by improbable-ludwik**
Catching up with official version 4.0.0 and adding our own output folder features.

### CODE REVIEW

**Comment by domgreen** [comment]
Im not sure what the changes are here ... is it updating to the latest version of the plugin only? Or are we adding extra stuff

**Comment by improbable-ludwik** [comment]
@domgreen  Our fork was made under a previous version where a small feature was added. This brings us up to v 4.0.0 and apply previous enhancements made.

**Comment by improbable-ludwik** [comment]
@domgreen @ianbillett @bwplotka  Please take another look. The delta of this v4.0.1 vs the official v4.0.0 is quite small as per here: https://github.com/logstash-plugins/logstash-output-google_cloud_storage/compare/master...improbable:CDATA-560-logstash-gcp-folder

File: lib/logstash/outputs/gcs/path_factory.rb [hunk 1]
```diff
@@ -80,7 +80,8 @@ def template_variables
               prefix: @prefix,
               host: Socket.gethostname,
               date: Time.now.strftime(@date_pattern),
-              partf: '%03d' % @part_number,
+              partf: '%03d' % @part_number,  ## This errors with  `:backtrace=>["org/jruby/RubyString.java:1249:in `%'", "/Users/ludwik/code/testing/logstash-7.0.1/vendor/local_gems/23c3f86a/logstash-output-google_cloud_storage-4.0.1-java/lib/logstash/outputs/gcs/path_factory.rb:83:in `template_variables'"`
```
Comment (domgreen) [inline lib/logstash/outputs/gcs/path_factory.rb]: should this be `/Users/ludwik...`?

File: lib/logstash/outputs/google_cloud_storage.rb [hunk 2]
```diff
@@ -53,9 +54,8 @@
 # output {
 #    google_cloud_storage {
 #      bucket => "my_bucket"                                     (required)
-#      key_path => "/path/to/privatekey.p12"                     (required)
-#      key_password => "notasecret"                              (optional)
-#      service_account => "1234@developer.gserviceaccount.com"   (required)
+#      folder => "my_prepended_bucker_folder"                    (optional)
```
Comment (domgreen) [inline lib/logstash/outputs/google_cloud_storage.rb]: now required?

**Pull request merged**